### PR TITLE
Clustering policytype list

### PR DESCRIPTION
**Pull request by dkt26111**
For #823 

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

API doc:
https://developer.openstack.org/api-ref/clustering/#list-policy-types

API code:
https://github.com/openstack/senlin/blob/master/senlin/api/openstack/v1/policy_types.py#L30

### CODE REVIEW

**Comment by dkt26111** [comment]
@jtopjian: Requesting code review.  I'm working with JackKuei on implementing the senlin APIs.

File: openstack/clustering/v1/policytypes/results.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+// PolicyType represents a clustering policy type in the Openstack cloud
+type PolicyType struct {
+	Name          string                         `json:"name"`
+	Version       string                         `json:"version"`
+	SupportStatus map[string][]SupportStatusType `json:"support_status,omitempty"`
+}
+
+// SupportStatus represents the support status information for a clustering policy type
```
Comment (jtopjian) [inline openstack/clustering/v1/policytypes/results.go]: SupportStatusType?

File: openstack/clustering/v1/policytypes/urls.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package policytypes
+
+import "github.com/gophercloud/gophercloud"
+
+var apiVersion = "v1"
+var apiName = "policy-types"
```
Comment (jtopjian) [inline openstack/clustering/v1/policytypes/urls.go]: nit: these could be constants:

```go
const (
  apiVersion = "v1"
  apiName = "policy-types"
)
```

File: openstack/clustering/v1/policytypes/results.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import (
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// PolicyType represents a clustering policy type in the Openstack cloud
+type PolicyType struct {
+	Name          string                         `json:"name"`
+	Version       string                         `json:"version"`
+	SupportStatus map[string][]SupportStatusType `json:"support_status,omitempty"`
```
Comment (jtopjian) [inline openstack/clustering/v1/policytypes/results.go]: nit: `omitempty` isn't really needed here - it's only useful in requests.

**Comment by jtopjian** [comment]
@dkt26111 whoops - actually, my mistake. I was looking at policy list and not policy type list :)

File: openstack/clustering/v1/policytypes/results.go (hunk 1 above)
Comment (dkt26111) [reply openstack/clustering/v1/policytypes/results.go]: Yes, you are correct.  I will fix this.

File: openstack/clustering/v1/policytypes/urls.go (hunk 2 above)
Comment (dkt26111) [reply openstack/clustering/v1/policytypes/urls.go]: Agree.  I'll fix it.

File: openstack/clustering/v1/policytypes/results.go (hunk 3 above)
Comment (dkt26111) [reply openstack/clustering/v1/policytypes/results.go]: I'll fix that too.

**Comment by dkt26111** [comment]
If everything looks good to you in the latest commits, you can go ahead and merge it.  I will work with Jackkuei to rebase #824 .

**Comment by jtopjian** [comment]
@dkt26111 All looks good! Really nice work on this.

A note on microversions: we haven't figured out a good way of handling them in Gophercloud yet. Basic microversion stuff like what you're testing here is totally fine.

However if there will be times when, for example, a version 1.5 request or response body is different than a 1.0 request/body (meaning the struct fields are different and/or incompatible), this is the area we're still looking into. We can discuss different options if we get into that situation with Senlin.

**Pull request merged**